Added Topology labels

friegger opened the pull request:
# Proposed Changes

Adds the possibility to set topology labels on the machine, volume and bucket pools.

The labels have to be provided as arguments to the `/<x>poollet` command. The arguments are only considered if a non empty value is provided.

They are initially set when the `<x>Pool` is created. The reconciler ensures that they are reset in case they have been changed and only then.

Fixes #1411

friegger committed:
1411 Topology Labels

Adds the possibility to set topology labels on the machine pools.

The labels have to be provided as arguments to the `/machinepoollet`
command.

They are initially set when the `MachinePool` is created. The reconceiler
ensures they are reset in case they have been changed.

friegger committed:
Add reuse information

friegger committed:
Readd volumesnapshots and generate with latest binaries

balpert89 commented on api/common/v1alpha1/common_types.go:
```diff
@@ -15,6 +15,20 @@ import (
 
 const (
+	// TopologyRegionLabel is a label that can be applied to MachinePool, VolumePool, and BucketPool
+	//
```
nitpick: can you remove the extra linebreak, this renders the package documentation a bit weird.

friegger committed:
Use a map for the labels

friegger committed:
Merge remote-tracking branch 'origin/main' into feat/1411-add-topology-labels

friegger committed:
Remove unused helper fn

friegger committed:
Update docs

friegger committed:
Remove obsolete fields

friegger committed:
Add volume support

friegger committed:
Extract setting the labels to the common utils package

friegger committed:
Add topology labels also to buckets

friegger committed:
Add missing license info

friegger committed:
Inline labelschanged into if statement

friegger committed:
Remove unnecessary assignment

friegger committed:
Merge branch 'main' into feat/1411-add-topology-labels

friegger committed:
Apply review comments

Don't check for changes, but simply apply the patch, accepting
a potentially empty patch call.
